service/history/visibility_queue_task_executor.go

```diff
@@ -461,6 +461,12 @@ func (t *visibilityQueueTaskExecutor) cleanupExecutionInfo(
 	executionInfo.Memo = nil
 	executionInfo.SearchAttributes = nil
 	executionInfo.RelocatableAttributesRemoved = true
+
+	if t.shardContext.GetConfig().EnableUpdateClosedWorkflowByMutation() {
+		return weContext.UpdateWorkflowExecutionAsPassive(ctx, t.shardContext)
```

Comment:
Curious why `UpdateWorkflowExecutionAsPassive` here and HSM state replicator. Why not `UpdateWorkflowExecutionAsActive`?

Reply:
HSM/Activity state replicator are replication logic for applying changes in the standby cluster of a namespace. And when "AsPassive" is used, the mutable state logic will prevent another replication task from being generated (there are some other differences, but this is the most important one).

For visibility, you can think this the removal of Memo & SearchAttributes is a cluster local operation (move the data from mutable state to ES).  and this operation doesn't need to be replicated.